src/mvba/consensus.rs

```diff
@@ -88,8 +89,13 @@ impl Consensus {
             vcbc::MODULE_NAME => match bundle.target {
                 Some(target) => match self.vcbc_map.get_mut(&target) {
                     Some(vcbc) => {
-                        let msg = bincode::deserialize(&bundle.payload)?;
-                        vcbc.receive_message(bundle.initiator, msg, &mut self.broadcaster)?;
+                        let msg = match &bundle.message {
+                            VcbcMsg(msg) => msg,
+                            _ => {
+                                return Err(Error::InvalidMessage("unexpected message".to_string()))
```

Comment:
Yeah, if you pattern match on `bundle.message` instead of `bundle.module`, you no longer have this error checking

Reply:
True, I forgot it. Thanks for reminding